projects/fsastorefrontlib/src/core/sales-illustration/facade/sales-illustration-chart.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { TranslationService } from '@spartacus/core';
+import { EChartsOption } from 'echarts';
+import { combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { getSalesIllustrationChartConfig } from '../../chart-config/sales-illustration-chart-config';
+
+@Injectable()
+export class SalesIllustrationChartService {
+  constructor(private translationService: TranslationService) {}
+
+  getChartOptions(
+    salesIllustrationDiagramData: any
+  ): Observable<EChartsOption> {
+    const labelTranslations = [
+      this.translationService.translate('salesIllustration.expectedSavings'),
+      this.translationService.translate('fscommon.contribution'),
+      this.translationService.translate('salesIllustration.interest'),
```

Comment:
These should also come from some config, and passed to these `this.translationService.translate` properties. As it is now, it's fixed to Savings only.